pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -56,20 +56,8 @@ type AttachMetadata struct {
 	Node *bool `json:"node,omitempty"`
 }
 
-// EC2Filter is the configuration for filtering EC2 instances.
-type EC2Filter struct {
-	Name   string   `json:"name"`
-	Values []string `json:"values"`
-}
-
-// DockerFilter is the configuration to limit the discovery process to a subset of available resources.
-type DockerFilter struct {
-	Name   string   `json:"name"`
-	Values []string `json:"values"`
-}
-
 // Filter is the configuration to limit the discovery process to a subset of available resources.
-type DockerSwarmFilter struct {
+type Filter struct {
 	// Name is the key of the field to check against.
```

Comment:
Since the struct is used for different SDs, let's have a generic description. 
```suggestion
	// Name of the filter.
```

And for the Values field

````
// Values to filter on.
````